lib/automate/modules/send.py

```diff
@@ -14,26 +16,55 @@ def __init__(self):
 
     def run(self, to, when, body, sender):
         user, _ = fuzzy.extractOne(sender, SETTINGS["users"].keys())
-        settings = SETTINGS["users"][user]["email"]
-        username = settings.get("username")
-        password = settings.get("password")
-        content = body + f"\n\nRegards,\n{user}"
+        self.settings = SETTINGS["users"][user]["email"]
+        self.username = self.settings.get("username")
+        self.password = self.settings.get("password")
+        self.subject = body.partition("\n")[0]
+        self.content = body + f"\n\nRegards,\n{user}"
 
+        if len(to) == 1:
+            receiver = to[0]
+        elif len(to) == 0:
+            raise NoReceiverError("No receiver was entered")
+        else:
+            raise ToManyReceiversError("Can only handle one (1) receiver at this time")
+
+        if not self.is_email(receiver):
+            # filter out the contacts that does not need to be considered
+            possible_receivers = fuzzy.extract(receiver, SETTINGS["users"].keys())
+            possible_receivers = list(filter(lambda x: x[1] > 75, possible_receivers))
```

Comment:
You should probably remove the sender here as well, you rarely want to send an email to yourself (I only do it for primitive file sharing between computers, and that probably shouldn't be considered.).

Reply:
I can filter it out for now, we should probably split the local contact book and the information about the user but this is solved when using an external contact book. You can go ahead and look into the external if you want to